R/mobr_boxplots.R

```diff
@@ -125,24 +124,105 @@ get_pval <- function(rand, obs, n_samples)
    return(p_val)
 }
 
+# Get group-level biodiversity indices
+calc_biodiv_groups <- function(abund_mat, groups, index, n_rare)
+{
+   out <- list("group" = groups)
+   
+   # Number of individuals -----------------------------------------------------
+   if ("N" %in% index){
+      out$N  = rowSums(abund_mat)     
+   } 
+   
+   # Number of species ---------------------------------------------------------
+   if ("S" %in% index){
+      out$S  = rowSums(abund_mat > 0)  
+   }  
+   
+   # Rarefied richness ---------------------------------------------------------
+   if ("S_rare" %in% index){  
+      
+      out$S_rare <- list()
+      
+      n_rare <- floor(n_rare)
+      if (is.na(n_rare)){   
```

Comment:
I'm guessing that `n_rare` is a vector. When `is.na()` is applied to a vector, it returns a binary vector of TRUE/FALSE, and the if statement may not work as intended.